tests/InferTypesFuzz.elm

```diff
@@ -0,0 +1,329 @@
+module InferTypesFuzz exposing (exprTyped)
+
+import AST.Canonical as Canonical
+import AST.Common.Literal exposing (Literal(..))
+import AST.Common.Located as Located
+import AST.Common.Type as Type exposing (Type)
+import Common.Types exposing (VarName(..))
+import Fuzz exposing (Fuzzer)
+import Random exposing (Generator)
+import Random.Extra as Random
+import Shrink exposing (Shrinker)
+import TestHelpers exposing (located)
+
+
+exprTyped : Type -> Fuzzer Canonical.LocatedExpr
+exprTyped targetType =
+    Fuzz.custom
+        (exprGenerator targetType)
+        exprShrinker
+
+
+exprGenerator : Type -> Generator Canonical.LocatedExpr
+exprGenerator targetType =
+    let
+        cannotFuzz details =
+            let
+                prefix =
+                    "Cannot fuzz `" ++ dumpType targetType ++ "` expressions."
+
+                message =
+                    if String.isEmpty details then
+                        prefix
+
+                    else
+                        prefix ++ " " ++ details
+            in
+            Debug.todo message
+    in
+    case targetType of
+        Type.Int ->
+            intExpr
+
+        Type.Float ->
+            floatExpr
+
+        Type.Bool ->
+            boolExpr
+
+        Type.Char ->
+            charExpr
+
+        Type.String ->
+            stringExpr
+
+        Type.Unit ->
+            unitExpr
+
+        Type.List elementType ->
+            if Type.isParametric elementType then
+                cannotFuzz "Only lists with non-parametric element types are supported."
```

Comment:
we could probably return an empty list here?

Reply:
The empty list is inferred to what dump as `List a`. We could generate `Canonical.List []` for `List (Var n)`, but I'm not certain it will always work out well with recursive generation. It might be ok but I have not thought it through.

Reply:
Ah. Right, let's keep it as it is.